web/src/settings_components.ts

```diff
@@ -357,6 +357,73 @@ function get_message_retention_setting_value(
     return check_valid_number_input(custom_input_val);
 }
 
+type FieldData = Record<string, Record<string, string> | string>;
```

Comment:
Similar to [ProfileFieldData](https://github.com/zulip/zulip/blob/326944542ee903416b8ec988cf264768a59c3d9f/zerver/lib/types.py#L50): `Dict[str, Union[Dict[str, str], str]]`